src/compiler/TypescriptCompiler.ts

```diff
@@ -29,10 +31,16 @@ export class TypescriptCompiler {
         private readonly ts: TypeScript,
         private readonly appValidator: AppsEngineValidator,
     ) {
-        this.compilerOptions = {
+        // Things we shouldn't allow the app dev changing
+        this.nonConfigurableCompilerOptions = {
```

Comment:
Why not? Is there a reason for this options to be hardcoded?

Reply:
Because these options affect the output of the compilation, and that's up to the host system to decide, not the app developer